zkevm-circuits/src/exp_circuit.rs

```diff
@@ -25,9 +25,21 @@ use halo2_proofs::{
     plonk::{ConstraintSystem, Error, Selector},
     poly::Rotation,
 };
-use param::*;
+
+use crate::evm_circuit::util::constraint_builder::ConstrainBuilderCommon;
 use std::{marker::PhantomData, ops::Add};
 
+/// The number of rows assigned for each step in an exponentiation trace.
+pub const OFFSET_INCREMENT: usize = 7usize;
+/// The number of rows required for the exponentiation table within the circuit
+/// for each step.
+pub const ROWS_PER_STEP: usize = 4usize;
+/// The gate "verify all but the last step" at constraint "`base_limb[i]` is the
+/// same across all steps" uses rotation 10 in `exp_table.base_limb` which is
+/// enabled with `q_usable`, which in turn is enabled in all steps.  This means
+/// this circuit requires these extra rows after the last enabled `q_usable`.
+const UNUSABLE_EXP_ROWS: usize = 10usize;
+
```

Comment:
I think this got slipped away during the rebase with main.  These constants are now in `use param::*`

Reply:
@ed255 fixed, check again please